plugins/device/rest.go

```diff
@@ -194,3 +194,82 @@ func deviceEditHandler(w http.ResponseWriter, r *http.Request) {
 
 	restutils.SendHTTPResponse(ctx, w, http.StatusOK, nil)
 }
+
+func deviceDeleteHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+	peerID := mux.Vars(r)["peerid"]
+	if uuid.Parse(peerID) == nil {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, "invalid peer-id passed in url")
+		return
+	}
+	deviceName := mux.Vars(r)["device"]
+	if deviceName == "" {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, "Device name not provided in URL")
+		return
+	}
+
+	// Adding prefix(/) to device name
+	deviceName = "/" + deviceName
+
+	txn, err := transaction.NewTxnWithLocks(ctx, peerID+deviceName)
+	if err != nil {
+		status, err := restutils.ErrToStatusCode(err)
+		restutils.SendHTTPError(ctx, w, status, err)
+		return
+	}
+	defer txn.Done()
+
+	peerInfo, err := peer.GetPeer(peerID)
+	if err != nil {
+		logger.WithError(err).WithField("peerid", peerID).Error("Peer ID not found in store")
+		if err == errors.ErrPeerNotFound {
+			restutils.SendHTTPError(ctx, w, http.StatusNotFound, errors.ErrPeerNotFound)
+		} else {
+			restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, "failed to get peer details from store")
+		}
+		return
+	}
+
+	txn.Nodes = []uuid.UUID{peerInfo.ID}
+	txn.Steps = []*transaction.Step{
+		{
```

Comment:
@aravindavk  don't we need to check any volume present on the device before removing it?

Reply:
Vgremove will fail if the device is being used. We don't wipe the device

Reply:
no harm in checking any volume present on the device, so that user will get a meaning full message something like bricks present on the device, cannot delete the device,

Reply:
Done